Can't build current 0.16

```
$ go get github.com/lxc/lxd
# github.com/lxc/lxd/shared
go/src/github.com/lxc/lxd/shared/util_linux.go: In function ‘create_pipe’:
go/src/github.com/lxc/lxd/shared/util_linux.go:100:6: warning: implicit declaration of function ‘pipe2’ [-Wimplicit-function-declaration]
  if (pipe2(pipefd, O_CLOEXEC) < 0) {
      ^
```

--- a/Makefile
+++ b/Makefile
@@ -12,13 +12,15 @@ ARCHIVE=lxd-$(VERSION).tar
 
 .PHONY: default
 default:
-	go get -t -v -d ./... || true
+	-go get -t -v -d ./...
 	go install -v ./...
+	echo "LXD built succesfuly"
 
 .PHONY: client
 client:
-	go get -t -v -d ./...
+	-go get -t -v -d ./...
 	go install -v ./lxc
+	echo "LXD client built succesfuly"
 
 # This only needs to be done when migrate.proto is actually changed; since we
 # commit the .pb.go in the tree and it's not expected to change very often,
@@ -34,6 +36,7 @@ check: default
 
 gccgo:
 	go build -compiler gccgo ./...
+	echo "LXD built succesfuly with gccgo"
 
 .PHONY: dist
 dist:

--- a/shared/util_linux.go
+++ b/shared/util_linux.go
@@ -14,6 +14,7 @@ import (
 
 // #cgo LDFLAGS: -lutil -lpthread
 /*
+#define _GNU_SOURCE
 #include <sys/types.h>
 #include <unistd.h>
 #include <stdlib.h>

